feedserver/did.go

```diff
@@ -1,28 +1,45 @@
 package feedserver
 
 import (
+	"encoding/json"
 	"fmt"
-	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 	"net/http"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 )
 
 func serverDID(hostname string) string {
 	return fmt.Sprintf("did:web:%s", hostname)
 }
 
+func generateDIDJson(hostname string) ([]byte, error) {
+	type Object map[string]any
+
+	did := Object{
+		"@context": []string{"https://www.w3.org/ns/did/v1"},
+		"id":       serverDID(hostname),
+		"service": []Object{{
+			"id":              "#bsky_fg",
+			"type":            "BskyFeedGenerator",
+			"serviceEndpoint": fmt.Sprintf("https://%s", hostname),
+		}},
+	}
+
+	return json.Marshal(did)
+}
+
 func didHandler(hostname string) (string, http.Handler) {
+	did, err := generateDIDJson(hostname)
+
+	if err != nil {
+		panic(fmt.Errorf("could not generate did.json: %v", err))
```

Comment:
I'd prefer to avoid panicking here. We have two choices:
- Return this error wrapped, and then handle it in `New`. If we need to make `New` return an error, that's fine.
- Push the `json.Marshal` down to the actual handler func body, and call `handleErr` on it. This is a bit less efficient, we don't need to marshal this repeatedly if it's going to remain the same.

Reply:
Yeah I agree with not panicking. I’ve opted for returning the error with `New`; not for performance reasons but so it’ll fail early if it can’t marshal the value.

Reply:
Agreed - failing early is always better.